### PR TITLE
Fix for edit preferences not persisting

### PR DESCRIPTION
**Pull request by misslivirose**
Addressing FB 7364. These values were previously set to true at startup of Interface instead of checking for their set values.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8227/app_12699_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8227/HighFidelity-Beta-PR12699-454a8a4aac7598fd7dcae48419c9300b62fcfae4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8227/HighFidelity-Beta-PR12699-454a8a4aac7598fd7dcae48419c9300b62fcfae4.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8238/app_12699_PR.apk

**Comment by misslivirose** [comment]
**Test Plan**
1. Install the build from this PR
2. Enabled 'Advanced' and 'Developer' Menus
3. Confirm that the following options are checked: Create Entities as Grabbable, Allow Selecting of Small, Allow Selecting of Large. Confirm that 'Allow Selecing of Lights" is off by default.
4. In Create mode, confirm that you can select small and large models, that you only can select a light by clicking the icon for it. Confirm that new items you make are grabbable.
5. In the Edit menu, toggle off the menu items and test the inverse of the above settings. 
6. Confirmed that while unchecked in Create mode, you do not create grabbable items, you cannot select small entities or large entities, and that you can select a light by clicking in the vicinity of the light icon.
7. Restart Interface.
8. Confirm that the menu items remain in their last known configuration and persist across a restart. Note that if Interface crashes on exit, these may not get saved. 
9. Confirm that if you reset your settings in Interface that the default checked options return.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8238/HighFidelity-Beta-PR12699-f115fd35c9a1d276fdbc16da0ba5b897e70f4807.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8238/HighFidelity-Beta-PR12699-f115fd35c9a1d276fdbc16da0ba5b897e70f4807.exe

**Comment by conklin94122** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8276/app_12699_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8276/HighFidelity-Beta-PR12699-f115fd35c9a1d276fdbc16da0ba5b897e70f4807.dmg

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8293/app_12699_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8293/HighFidelity-Beta-PR12699-f115fd35c9a1d276fdbc16da0ba5b897e70f4807.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8293/HighFidelity-Beta-PR12699-f115fd35c9a1d276fdbc16da0ba5b897e70f4807.exe

File: scripts/system/edit.js [hunk 1]
```diff
@@ -1135,34 +1143,35 @@ function setupModelMenus() {
 
     Menu.addMenuItem({
         menuName: GRABBABLE_ENTITIES_MENU_CATEGORY,
-        menuItemName: GRABBABLE_ENTITIES_MENU_ITEM,
+        menuItemName: MENU_CREATE_ENTITIES_GRABBABLE,
         afterItem: "Unparent Entity",
         isCheckable: true,
-        isChecked: true,
+        isChecked: Settings.getValue(SETTING_EDIT_PREFIX + MENU_CREATE_ENTITIES_GRABBABLE) !== false,
```
Comment (thoys) [inline scripts/system/edit.js]: Settings.getValue   has an optional default value that you could use and prevent the !== false situation

File: scripts/system/edit.js [hunk 2]
```diff
@@ -1135,34 +1143,35 @@ function setupModelMenus() {
...
 
     Menu.addMenuItem({
         menuName: "Edit",
-        menuItemName: "Allow Selecting of Large Models",
-        afterItem: GRABBABLE_ENTITIES_MENU_ITEM,
+        menuItemName: MENU_ALLOW_SELECTION_LARGE,
+        afterItem: MENU_CREATE_ENTITIES_GRABBABLE,
         isCheckable: true,
-        isChecked: true,
+        isChecked: Settings.getValue(SETTING_EDIT_PREFIX + MENU_ALLOW_SELECTION_LARGE) !== false,
```
Comment (thoys) [inline scripts/system/edit.js]: Settings.getValue   has an optional default value that you could use and prevent the !== false situation

File: scripts/system/edit.js [hunk 3]
```diff
@@ -1135,34 +1143,35 @@ function setupModelMenus() {
...
     });
     Menu.addMenuItem({
         menuName: "Edit",
-        menuItemName: "Allow Selecting of Small Models",
-        afterItem: "Allow Selecting of Large Models",
+        menuItemName: MENU_ALLOW_SELECTION_SMALL,
+        afterItem: MENU_ALLOW_SELECTION_LARGE,
         isCheckable: true,
-        isChecked: true,
+        isChecked: Settings.getValue(SETTING_EDIT_PREFIX + MENU_ALLOW_SELECTION_SMALL) !== false,
```
Comment (thoys) [inline scripts/system/edit.js]: Settings.getValue   has an optional default value that you could use and prevent the !== false situation

File: scripts/system/edit.js [hunk 4]
```diff
@@ -1135,34 +1143,35 @@ function setupModelMenus() {
...
         grouping: "Advanced"
     });
     Menu.addMenuItem({
         menuName: "Edit",
-        menuItemName: "Allow Selecting of Lights",
-        afterItem: "Allow Selecting of Small Models",
+        menuItemName: MENU_ALLOW_SELECTION_LIGHTS,
+        afterItem: MENU_ALLOW_SELECTION_SMALL,
         isCheckable: true,
+        isChecked: Settings.getValue(SETTING_EDIT_PREFIX + MENU_ALLOW_SELECTION_LIGHTS),
```
Comment (thoys) [inline scripts/system/edit.js]: Settings.getValue   has an optional default value, which should be a boolean in this case

**Comment by misslivirose** [comment]
Woah. I think I messed up in updating my local dev environment here. Not sure why it picked up all these extra changes.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8402/app_12699_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8404/app_12699_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8404/HighFidelity-Beta-PR12699-799963a00905e7115feb84a9e4ee962d4dbf2033.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8404/HighFidelity-Beta-PR12699-799963a00905e7115feb84a9e4ee962d4dbf2033.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Fail @conklin94122 
After a restart, the Create Entities as Grabbable option in the edit menu is no longer checked. After a crash and settings are reset, the Create Entities as Grabbable option is checked. Logs and video attached below. Other 2 options listed in test plan do remain checked after a reset.
[Files.zip](https://github.com/highfidelity/hifi/files/1868652/Files.zip)

**Comment by misslivirose** [comment]
I don't see any menu items in the video, it's not clear to me which part of this is a fail. Can you please elaborate on the test steps and where you saw something unexpected?

**Comment by qlimnavarro** [comment]
@misslivirose Sorry about that, not sure why the video didnt get that. After a restart, the Create Entities as Grabbable option in the edit menu does not save its last known configuration. Upon opening the interface after a fresh install, the Create Entities as Grabbable option is checked. Once after that it will be unchecked by default when the interface is opened.
[Video.zip](https://github.com/highfidelity/hifi/files/1895722/Video.zip)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8668/app_12699_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8668/HighFidelity-Beta-PR12699-c7be6e09346618542f52ba503a992eaf85e1ffea.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8668/HighFidelity-Beta-PR12699-c7be6e09346618542f52ba503a992eaf85e1ffea.exe

**Comment by thoys** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8671/app_12699_PR.apk

**Comment by marko8904** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8676/HighFidelity-Beta-PR12699-c7be6e09346618542f52ba503a992eaf85e1ffea.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8676/HighFidelity-Beta-PR12699-c7be6e09346618542f52ba503a992eaf85e1ffea.exe

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8679/app_12699_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8679/HighFidelity-Beta-PR12699-c7be6e09346618542f52ba503a992eaf85e1ffea.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Menu items now remember their last configuration.

**Pull request merged**